test/parallel/test-http-information-processing.js

```diff
@@ -2,22 +2,21 @@
 require('../common');
 const assert = require('assert');
 const http = require('http');
-const Countdown = require('../common/countdown');
 
-const test_res_body = 'other stuff!\n';
-const countdown = new Countdown(3, () => server.close());
+const testResBody = 'other stuff!\n';
+const kMessageCount = 2;
 
 const server = http.createServer((req, res) => {
-  console.error('Server sending informational message #1...');
-  res.writeProcessing();
-  console.error('Server sending informational message #2...');
-  res.writeProcessing();
+  for (let i = 0; i < kMessageCount; i++) {
+    console.error(`Server sending informational message #${i}...`);
+    res.writeProcessing();
+  }
   console.error('Server sending full response...');
   res.writeHead(200, {
     'Content-Type': 'text/plain',
     'ABCD': '1'
   });
-  res.end(test_res_body);
+  res.end(testResBody);
```

Comment:
`common.mustCall` variant can be used here.

Reply:
Please, no. We really don't need to put `common.mustCall()` on everything.

Reply:
We could, but I'm not sure we should. This test isn't checking to see if `res.end()` fires it's callback or not. If the response isn't sent, the test will fail. We don't need the extra check. Adding checks that are unrelated to the test case makes it harder for someone reading the code to understand what the test is actually supposed to be testing.

This makes it sound like I'm much more opposed to the idea than I actually am. I'm on the fence, around a -0 or so.

Reply:
> This test isn't checking to see if res.end() fires its callback or not

correct me if I am wrong, but isn't that the case with most of 100+ http test cases?
 - intent of the test does not cover explicit checking on `res.end`
 - but as a matter of practice, they do that check
if you can generalize a statement around `whether http tests should check life cycle events are met or not` I can also work towards implementing this across other tests too.

Reply:
> * intent of the test does not cover explicit checking on `res.end`
> * but as a matter of practice, they do that check

I'm not seeing that with `res.end()` calls. I'm using `grep` and doing a naive/superficial check, so results may be imperfect but close enough to draw reasonable conclusions:

* Almost none of the 200+ calls to `res.end()` in `parallel/test-http*` have a callback wrapped in `mustCall()`. The majority of them are like the call here where `res.endI()` is called and no callback is provided.
* Of those that do wrap a callback in `muscCall()`, only one file checks empty callback: `test-http-outgoing-end-multiple.js`. That makes sense for that test because it is literally checking that calling `res.end()` multiple times will work as expected.

I don't think we should add a check for a callback here that we're not using. Again, I'm not strongly opposed, but I am mildly resistant. 😀

Reply:
I used this command `grep -e "res.on('end" -e "response.on('end" test-http*.js | grep "mustCall" | wc -l` and got around 50 instances. Unfortunately I don't have the insight to validate their context. so I will leave it at that. Thank you!